scalafix-core/src/main/scala/scalafix/rewrite/NoValInForComprehension.scala

```diff
@@ -0,0 +1,21 @@
+package scalafix
+package rewrite
+
+import scala.meta._
+import scala.meta.contrib._
+
+case object NoValInForComprehension extends Rewrite {
+
+  override def rewrite(ctx: RewriteCtx): Patch = {
+    ctx.tree.collect {
+      case f: Term.ForYield =>
```

Comment:
Oh, ok, I'm ufficially confused by the grammar.

I tried

```scala
(for { x <- List(1, 2, 3) })
```

and the parser complains. By looking at the grammar:


```
`for' (`(' Enumerators `)' | `{' Enumerators `}') {nl} [`yield'] Expr
```

`Expr` is mandatory, so something like


```scala
(for { x <- List(1, 2, 3) } 42)
```

works instead (the type of the expression is `Unit`). I'm wondering why `Expr` is needed without a `yield`...